authentication/auth.go

```diff
@@ -0,0 +1,86 @@
+package authentication
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/skratchdot/open-golang/open"
+	"golang.org/x/oauth2"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+var (
+	conf      *oauth2.Config
+	ctx       context.Context
+	verifier  string
+	authToken string
+)
+
+func callbackHandler(w http.ResponseWriter, r *http.Request) {
+	queryParts, _ := url.ParseQuery(r.URL.RawQuery)
+
+	code := queryParts["code"][0]
+
+	tok, err := conf.Exchange(ctx, code, oauth2.VerifierOption(verifier))
+	if err != nil {
+		log.Fatal(err)
+	}
+	authToken = tok.AccessToken
+	_, err = io.WriteString(w, "Login successful! Please close this window and return back to CLI")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func serve(done chan int) {
+	log.Fatal(http.ListenAndServe(":9998", nil))
+	if done != nil {
+		done <- 0
+	}
+}
+
+func VerifyLogin(clientID string, clientSecret string) (string, error) {
+	authToken = ""
+	ctx = context.Background()
+	conf = &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
```

Comment:
I don't believe we'll need to provide Client Secret for this flow. Can you confirm with Josh?

Reply:
Removed client secret from input params